app/src/main/java/de/danoeh/antennapod/dialog/FilterDialog.java

```diff
@@ -58,5 +69,188 @@ public void openDialog() {
         builder.create().show();
     }
 
+    public void openCustomDialog() {
+
+        final String[] values = context.getResources().getStringArray(R.array.episode_filter_values);
+
+        final Set<String> filterValues = new HashSet<>(Arrays.asList(filter.getValues()));
+        final Set<String> selectedValues = new HashSet<>();
+        final String[] checkedItems = new String[values.length];
+
+        for (int i = 0; i < values.length; i++) {
+            String value = values[i];
+            if (filterValues.contains(value)) {
+                checkedItems[i] = value;
+            }
+        }
+
+        final String[] selection1 = context.getResources().getStringArray(R.array.episode_filter_selection_text1);
+        final String[] selection2 = context.getResources().getStringArray(R.array.episode_filter_selection_text2);
```

Comment:
Ok, what if we didn't have to handle it at all?; which is what's going on in the openDialog method. Looks like even more unnecessary code to me. I intended the optionSwitch hack to be temporary because I assumed we could add values to the filter that were missing.

    <string-array name="episode_filter_values">
        <item>unplayed</item>
        <item>paused</item>

<item>not paused</item>
<item>not favorite</item>
<item>no media</item>

        <item>played</item>
        <item>queued</item>
        <item>not_queued</item>
        <item>downloaded</item>
        <item>not_downloaded</item>
        <item>has_media</item>
        <item>is_favorite</item>
    </string-array>

Why does downloaded have an off/on state but paused does not?

we just need "not paused", "not favorite" and "no media" added to the available options( i just don't know where i thought it was beyond the scope of this design) . If you want to add more later you just add an on and off state for each new option. "on new, off new"

It looks like what you're saying is you want to get rid of "episode_filter_values" array and come up with something new based on an enum. I understand deflating the rows, designing in android studio, more maintainable etc. ok

Reply:
> If you want to add more later you just add an on and off state for each new option. "on new, off new"

There are options that are 3-state, for example New/Unplayed/Played. New is currently not filterable (would clutter the UI) but I want to keep the possibility to do that. That's why I do not really like the arrays that need to be kept in sync and only allow 2 states. Having to keep 4 arrays in sync (keys+translations for 2 columns) sounds like it will cause trouble later.

> the value being used to display text is also the value of the option if you look at openDialog()

Yeah, that's a problem for translation. My enum approach would allow to specify ID (for saving) and UI string separately.

Reply:
ok, sounds good

Reply:
okay, i've updated the pull request